src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/CopyNumberTriStateTransitionProbabilityCache.java

```diff
@@ -1,34 +1,45 @@
-package org.broadinstitute.hellbender.utils.hmm;
+package org.broadinstitute.hellbender.tools.exome.germlinehmm;
 
 import com.google.common.annotations.VisibleForTesting;
 import org.apache.commons.math3.linear.*;
+import org.broadinstitute.hellbender.utils.Utils;
 import org.broadinstitute.hellbender.utils.param.ParamUtils;
 
+import java.io.Serializable;
 import java.util.Arrays;
-import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
```

Comment:
We do. I discovered after a lot of confusing errors. When running the forward-backward algorithm concurrently on multiple samples (via [...].stream().parallel().[...]), they need to be able to concurrently cache log transition probabilities. Concurrent writing on a non-concurrent map produces garbage on collisions.